custom_components/unifiprotect/config_flow.py

```diff
@@ -36,79 +49,123 @@ class UnifiProtectFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     @staticmethod
     @callback
-    def async_get_options_flow(config_entry):
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
         """Get the options flow for this handler."""
         return OptionsFlowHandler(config_entry)
 
-    async def async_step_user(self, user_input=None):
-        """Handle a flow initiated by the user."""
-        if user_input is None:
-            return await self._show_setup_form(user_input)
-
-        errors = {}
+    @callback
+    async def _async_create_entry(self, title: str, data: Dict[str, Any]):
+        return self.async_create_entry(
+            title=title,
+            data={**data, CONF_ID: title},
+            options={
+                CONF_DISABLE_RTSP: False,
+                CONF_DOORBELL_TEXT: "",
+            },
+        )
 
+    @callback
+    async def _async_get_nvr_data(
+        self,
+        user_input: Dict[str, Any],
+        entry: Optional[config_entries.ConfigEntry] = None,
+    ) -> Tuple[Optional[Dict[str, Any]], Dict[str, str]]:
         session = async_create_clientsession(
             self.hass, cookie_jar=CookieJar(unsafe=True)
         )
 
-        unifiprotect = UpvServer(
-            session,
-            user_input[CONF_HOST],
-            user_input[CONF_PORT],
-            user_input[CONF_USERNAME],
-            user_input[CONF_PASSWORD],
+        if CONF_HOST in user_input:
+            host = user_input[CONF_HOST]
+            port = user_input.get(CONF_PORT, DEFAULT_PORT)
+        # reauth flow, pull host/port from existing settings
+        elif entry:
+            host = entry.data[CONF_HOST]
+            port = entry.data.get(CONF_PORT, DEFAULT_PORT)
+        else:
+            return None, {}
+
+        protect = UpvServer(
+            session=session,
+            host=host,
+            port=port,
+            username=user_input[CONF_USERNAME],
+            password=user_input[CONF_PASSWORD],
         )
 
+        errors = {}
+        nvr_data = None
         try:
-            server_info = await unifiprotect.server_information()
-            if server_info["server_version"] < MIN_REQUIRED_PROTECT_V:
+            nvr_data = await protect.server_information()
+            if nvr_data["server_version"] < MIN_REQUIRED_PROTECT_V:
                 _LOGGER.debug("UniFi Protect Version not supported")
                 errors["base"] = "protect_version"
-                return await self._show_setup_form(errors)
-
         except NotAuthorized as ex:
             _LOGGER.debug(ex)
             errors["base"] = "connection_error"
```

Comment:
```suggestion
            errors[CONF_PASSWORD] = "connection_error"
```

This is actually the field were the error will show up.   `connection_error` 

We should use the standard `invalid_auth` here so we can use the following in `strings.json`

```
"invalid_auth": "[%key:common::config_flow::error::invalid_auth%]"
```